src/dune/dune.py

```diff
@@ -531,6 +531,7 @@ def get_table(self, acnt, scope, tab):
     def features():
         return ["KV_DATABASE",
                 "ACTION_RETURN_VALUE",
+                "CONFIGURABLE_WASM_LIMITS2",
                 "BLOCKCHAIN_PARAMETERS",
                 "GET_SENDER",
                 "FORWARD_SETCODE",
```

Comment:
This list does not look correct. `KV_DATABASE` is an old eosio feature never ported to Leap.
Missing some as well, for example, `GET_CODE_HASH`.
The complete list can be found here: `libraries/chain/protocol_feature_manager.cpp`.

Reply:
> This list does not look correct. `KV_DATABASE` is an old eosio feature never ported to Leap. Missing some as well, for example, `GET_CODE_HASH`. The complete list can be found here: `libraries/chain/protocol_feature_manager.cpp`.

@heifner thank you very much for your advice  I found that success of deployment of eosio.system depends on speed of the hardware platform. You can see how it looks in the first part of the log in the description. With the timeout_exception. How do you think is this ok if the system can't bootstrap on some platforms because they are not fast enough or I should create an issue about this problem?

Reply:
Typically a re-try of the eosio.system deployment will work. You should add auto-retry of deployment on failure. That is true for all cleos commands. They should have a set number of auto-retries before considering it a true failure.

Reply:
> Typically a re-try of the eosio.system deployment will work. You should add auto-retry of deployment on failure. That is true for all cleos commands. They should have a set number of auto-retries before considering it a true failure.

I have tested the bootstrap sequence of DUNE (with this PR and added retries) and of bios-boot-tutorial.py on Phenom II X4 955 and Ryzen 7 5800H CPUs. Both sequences are not working correctly on Phenom CPU because of timeouts despite the fact the I increased timeouts in the config.ini in 10 times and added 15 retires for each command to DUNE. But both work excellent on Ryzen CPU even without retries in DUNE and increasing of timeouts in config.ini. Thus I think Phenom is just too slow CPU for correct work of EOS system.

In attachment log files of DUNE and bios-boot-tutorial.py on Phenom, and bios-boot-tutorial.py on Ryzen worked in identical Docker containers.

[log of DUNE on Phenom.txt](https://github.com/AntelopeIO/DUNE/files/10414929/log.of.DUNE.on.Phenom.txt)

[log of bios-boot-tutorial.py.txt](https://github.com/AntelopeIO/DUNE/files/10414937/log.of.bios-boot-tutorial.py.txt)

[bootstrap_successful_Ryzen.txt](https://github.com/AntelopeIO/DUNE/files/10414971/bootstrap_successful_Ryzen.txt)

Reply:
> This list does not look correct. `KV_DATABASE` is an old eosio feature never ported to Leap. Missing some as well, for example, `GET_CODE_HASH`. The complete list can be found here: `libraries/chain/protocol_feature_manager.cpp`.

GET_CODE_HASH is not exists in the bootstrap sequence in bios-boot-tutorial.py is this sequence is not correct too? I just don't know which minimal set of features should be enabled for successful bootstrap.

I have removed KV_DATABASE from the sequence. Take a look please.

Reply:
GET_CODE_HASH should exist in the bootstrap sequence of bios-boot-tutorial.py https://github.com/AntelopeIO/leap/issues/604

In my mind we should activate all of them or provide some mechanism to indicate which ones you want.

Reply:
> GET_CODE_HASH should exist in the bootstrap sequence of bios-boot-tutorial.py [AntelopeIO/leap#604](https://github.com/AntelopeIO/leap/issues/604)
> 
> In my mind we should activate all of them or provide some mechanism to indicate which ones you want.

I have added all missing features to the bootstrap sequence of DUNE to this PR and asked @stephenpdeos assign https://github.com/AntelopeIO/leap/issues/604 to me to add the same features there. I will add here results of the tests with extended timeouts on AMD Phenom II tomorrow.